src/metric/sensors/recorder.cpp

```diff
@@ -85,9 +85,10 @@ Recorder::Recorder(trace::Trace& trace)
         int chip_name_size = sensors_snprintf_chip_name(nullptr, 0, chip);
         assert(chip_name_size > 0);
 
-        std::string chip_name;
-        chip_name.resize(chip_name_size);
-        int res = sensors_snprintf_chip_name(chip_name.data(), chip_name.size(), chip);
+        std::vector<char> chip_name_vec(chip_name_size + 1);
+
+        int res = sensors_snprintf_chip_name(chip_name_vec.data(), chip_name_vec.size(), chip);
+        std::string chip_name(chip_name_vec.data());
```

Comment:
If you still want to use vector, move this line below the check, if `res` is less than 0, so you don't try to construct a string from possibly garbage data.